sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cra-integration/_tests_/cra-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,116 @@
+import { createMock, DeepMocked } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeCRAIncomeVerification,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import * as path from "path";
+import { CRAResponseIntegrationScheduler } from "../cra-response-integration.scheduler";
+import {
+  createFileFromStructuredRecords,
+  getStructuredRecords,
+  mockDownloadFiles,
+} from "@sims/test-utils/mocks";
+import { Job } from "bull";
+import { ApplicationStatus } from "@sims/sims-db";
+
+const CRA_FILENAME = "CRA_200_PBCSA00000.TXT";
+
+const padWithLeadingZeros = (num: number): string => {
+  // Pad the number with leading zeros to make it 9 digits long
+  return num.toString().padStart(9, "0");
+};
+
+describe(describeProcessorRootTest(QueueNames.CRAResponseIntegration), () => {
+  let app: INestApplication;
+  let processor: CRAResponseIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let craResponseFolder: string;
+
+  beforeAll(async () => {
+    craResponseFolder = path.join(__dirname, "cra-receive-files");
+    process.env.CRA_RESPONSE_FOLDER = craResponseFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    processor = app.get(CRAResponseIntegrationScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it("Should process CRA response file to update the income verification record.", async () => {
```

Comment:
Please use "Should/when" for the test descriptions.

Reply:
By "Should/when" I meant something like "**Should** update the income verification records **when** processing CRA response file."